src/checkValidTargetFilters.ts

```diff
@@ -24,10 +22,10 @@ export function checkValidTargetFilters(options: Options): Promise<void> {
     if (options.except) {
       return reject(new FirebaseError("Cannot specify both --only and --except"));
     }
-    if (targetContainsFilter(["database", "storage", "hosting"])) {
+    if (targetContainsFilter(["database", "storage"])) {
```

Comment:
This logic seems outdated - shouldn't this trigger for all VALID_TARGETS except functions & hosting? 

alid targets are:   
"database",
  "storage",
  "firestore",
  "functions",
  "hosting",
  "remoteconfig",
  "extensions",

Reply:
I've revamped this file to be a little better to understand. Let me know what you think.